src/model.js

```diff
@@ -1,52 +1,81 @@
 
-// Stat the directory to ensure it exists
-fs.stat(dataDirPath, function (err, stats) {
-  if (err && err.errno === -2) {
-    err.message = `Data directory "${dataDirPath}" not found; ${err.message}`
-  }
-  if (err) throw err
-  console.log(`GeoJSON files will be read from: ${dataDirPath}`)
-})
+const _ = require('lodash')
 
 /**
  * Model constructor
  */
-function Model () {}
+function Model (koop = {}) {
+  const dataDir = koop.dataDir || process.env.DATA_DIR || './data'
+  this.dataDirPath = path.join(process.cwd(), dataDir)
+  this.log = koop.log
+  this.verifyPathExists(this.dataDirPath)
+}
+
+Model.prototype.verifyPathExists = function verifyPathExists (dataDirPath) {
+  // Stat the directory to ensure it exists
+  const that = this
+  fs.stat(dataDirPath, function (err) {
+    if (err && err.errno === -2) {
+      err.message = `Data directory "${dataDirPath}" not found; ${err.message}`
+    }
+
+    if (err) {
+      throw err
+    }
+
+    that.log.warn(`GeoJSON files will be read from: ${dataDirPath}`)
+  })
+}
 
 /**
  * Fetch data from source.  Pass result or error to callback.
  * This is the only public function you need to implement on Model
  * @param {object} express request object
  * @param {function} callback
  */
-Model.prototype.getData = function (req, callback) {
-  let filename = `${req.params.id}.geojson`
-  fs.readFile(`${dataDir}/${filename}`, (err, dataBuffer) => {
+Model.prototype.getData = function getData (req, callback) {
+  const filename = `${req.params.id}.geojson`
+  const filePath = `${this.dataDirPath}/${filename}`
+  const that = this
+  fs.readFile(filePath, (err, dataBuffer) => {
     if (err && err.errno === -2) {
       err.code = 404
-      err.message = `File "${filename}" not found.`
+      err.message = 'File not found'
+      that.log.error(`${err.message}: ${filePath}`)
     } else if (err) {
       err.code = 500
     }
-    if (err) return callback(err)
 
-    // translate the response into geojson
-    const geojsonStr = dataBuffer.toString()
-    const geojson = translate(JSON.parse(geojsonStr))
+    if (err) {
+      return callback(err)
+    }
 
-    // Cache data for 10 seconds at a time by setting the ttl or "Time to Live"
-    geojson.ttl = 10
+    try {
+      // translate the response into geojson
+      const geojsonStr = dataBuffer.toString()
+      const geojsonParsed = JSON.parse(geojsonStr)
+      const metadataCopy = geojsonParsed.metadata
+      const geojson = translate(geojsonParsed)
 
-    // Add metadata
-    geojson.metadata = geojson.metadata || {}
-    geojson.metadata.title = 'Koop GeoJSON'
-    geojson.metadata.name = filename
-    geojson.metadata.description = `GeoJSON from ${filename}`
-    callback(null, geojson)
+      // Cache data for 10 seconds at a time by setting the ttl or "Time to Live"
+      geojson.ttl = _.get(metadataCopy, 'ttl', 10)
+
+      const detectedGeometryType = geojson.metadata.geometryType
```

Comment:
Why do you need this variable?

Reply:
Sorry, I didn't see your initial reply, since the conversation was resolved.  I would just remove it - doesn't appear to be used for anything at this point.

Thanks for the work on this, it needed help.

Reply:
On line 67, the `geometryType` gets replaced via parent object `metadata`. That way it preserves the original value.

Possibly the tests were failing before adding that.

TBH, I would remove `translate` and document the expected format that's just passed through. That way, you avoid overwriting the `title`, `name`, `description` in `translate`.

Reply:
`translate` is required in case the geojson data is not wrapped in a feature collection, so we can't remove that without a breaking change.  Plus we lose some functionality.